### ISSUE TITLE
If statement

### ISSUE THREAD

**Issue by antonvolkoff** [choix/slang#1]
When `if` is executed it evaluates both cases (true and false). Which could cause problems in a future. It should be moved out of `Env` to `eval`

**Comment by antonvolkoff** [comment]
Here is an example of broken code:

```
slang > (if true (def x 1) (def x 2))
1
slang > x
2
```

**Comment by antonvolkoff** [comment]
it was fixed with PR #2 

``` clojure
slang > (if true (set x 1) (set x 2))
1
slang > x
1
```
